### DISCUSSION TITLE
Is there any way to support quic http/3 protocol in the ingress?

### DISCUSSION THREAD

**Discussion by uGiFarukh** [nginx/kubernetes-ingress#2866] [Q&A] [answered]
Hello.

Is there any way I could enable the support for quick http/3 protocol in the nginx inc ingress and force it to all tls ingress resources somehow? I am just wondering how far is the http/3 implementation inside nginx ingress have come and is it available to test?

Thank You.

**Comment by brianehlert** [comment, marked as answer]
This NGINX Ingress Controller project is pure NGINX.  Without additional components such as OpenResty or LUA.
That said; if NGINX supports it, it can be done with this project.
The question usually comes down to how the configuration is achieved at a particular moment in time.

One of the things that has been delaying http/3 (quic) is OpenSSL.
That said, this project does have customers who build their own images and use brotli or boring.

You can find the official NGINX quic work here: https://quic.nginx.org/

**Reply by uGiFarukh** [reply]
Any guide to compile custom nginx alpine images with brotli or boring? I have no clue where to start with this. Also, is there any ETA from openssl side that you know of?

**Reply by brianehlert** [reply]
The latest details are in the comments here: https://www.nginx.com/blog/our-roadmap-quic-http-3-support-nginx/
